### PR TITLE
[MM-59544] Upgrade to Electron v31.2.1

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
Upgrading to Electron v31.2.1 for the upcoming release. Two small changes were made:
- Electron removed the `PermissionRequestHandlerHandlerDetails` so I had to recreate it. I think we'll want to refactor this properly later
- `getIOCounters()` was removed from Chromium so it's removed here.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-59544

```release-note
Upgrade to Electron v31.2.1
```

### CODE REVIEW

File: src/main/diagnostics/steps/step7.performance.ts [hunk 1]
```diff
@@ -31,7 +31,6 @@ const run = async (logger: MainLogger): Promise<DiagnosticStepResponse> => {
                     path: heapSnapshotFilepath,
                     success: process.takeHeapSnapshot(heapSnapshotFilepath),
                 },
-                IOCounters: process.getIOCounters(),
```
Comment (M-ZubairAhmed) [inline src/main/diagnostics/steps/step7.performance.ts:34]: are we ok with not having I/O stats anymore?

Comment (devinbinnie) [reply src/main/diagnostics/steps/step7.performance.ts:34]: Unfortunately we don't have much choice, Chromium removed them on us.

**Review by M-ZubairAhmed — APPROVED** [review]

File: src/main/diagnostics/steps/step7.performance.ts (hunk 1 above)
Comment (M-ZubairAhmed) [reply src/main/diagnostics/steps/step7.performance.ts:34]: Could be security related hence got removed

Comment (marianunez) [reply src/main/diagnostics/steps/step7.performance.ts:34]: Curious, where are these stats stored?

Comment (devinbinnie) [reply src/main/diagnostics/steps/step7.performance.ts:34]: You can get them from the Help menu > Run diagnostics

**Review by marianunez — APPROVED** [review]
LGTM. @devinbinnie any special smoke tests we want to do around permissions requests given the change here?

**Comment by devinbinnie** [comment]
> LGTM. @devinbinnie any special smoke tests we want to do around permissions requests given the change here?

It's just a type change, we should be fine with our usual round of testing

**Comment by marianunez** [comment]
> > LGTM. @devinbinnie any special smoke tests we want to do around permissions requests given the change here?
> 
> It's just a type change, we should be fine with our usual round of testing

@yasserfaraazkhan heads up on this ^

**Comment by yasserfaraazkhan** [comment]
> > > LGTM. @devinbinnie any special smoke tests we want to do around permissions requests given the change here?
> > 
> > 
> > It's just a type change, we should be fine with our usual round of testing
> 
> @yasserfaraazkhan heads up on this ^

thank you @devinbinnie , @marianunez 
the e2e have passed. I installed the nightly build to test installation, adding server, posts, threads, uploads, notifications, downloads on Mac.

**Pull request merged**